### PR TITLE
Add "pdgraster"

### PR DESCRIPTION
**Pull request by mfisher87**
Depends on #26489

> [!IMPORTANT]
> Not compatible with Windows _yet_.

cc @rushirajnenuji

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rushirajnenuji** [comment]
Hi, confirming my interest to be a maintainer for this package, thanks!

**Comment by mfisher87** [comment]
FYI: Windows build will probably fail due to a dependency (`pdgstaging`) and/or this library itself being incompatible with windows. We'll fix that in a future release!

**Comment by mfisher87** [comment]
@conda-forge/help-python, ready for review!

File: recipes/pdgraster/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{% set version = "0.9.2" %}
+{% set sha256 = "eb8ca23c2c8f794b88c1a680cc1a671d5d722093c33bb142f5fecd5f2be43e4b" %}
+{% set build_number = 0 %}
```
Comment (ocefpaf) [inline recipes/pdgraster/meta.yaml:3]:
```suggestion
```

File: recipes/pdgraster/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+package:
+  name: "pdgraster"
+  version: {{ version }}
+
+source:
+  url: https://github.com/PermafrostDiscoveryGateway/viz-raster/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/pdgraster/meta.yaml:11]:
```suggestion
  sha256: eb8ca23c2c8f794b88c1a680cc1a671d5d722093c33bb142f5fecd5f2be43e4b
```

File: recipes/pdgraster/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: {{ build_number }}
```
Comment (ocefpaf) [inline recipes/pdgraster/meta.yaml:16]:
```suggestion
  number: 0
```

**Review by ocefpaf — COMMENTED** [review]
Excess jinja doesn't really help but it obfuscates the recipe.

**Comment by mfisher87** [comment]
Personally, as a recipe maintainer, I find having all the values that need to regularly change in one place. But if that goes against convention, I'm fine with whatever.

**Comment by ocefpaf** [comment]
> Personally, as a feedstock maintainer, I find having all the values that need to regularly change in one place to be helpful. But if that goes against convention, I'm fine with whatever.

We rely on bot commands for the updates. If the package metadata is well defined, you rarely will need to change those manually.

**Comment by mfisher87** [comment]
I have some feedstocks that don't work with bot commands for various reasons (https://github.com/conda-forge/quarto-feedstock/issues/14), and I expect this repository might be one of them, because it neither uses GitHub Releases nor PyPI releases.

**Comment by ocefpaf** [comment]
> FYI: Windows build failed due to a dependency (`pdgstaging`) and/or this library itself being incompatible with windows. We'll fix that in a future release!

The failure seems to be a missing log file:

```
import: 'pdgraster'
Traceback (most recent call last):
  File "C:\bld\pdgraster_1719259170079\test_tmp\run_test.py", line 2, in <module>
    import pdgraster
  File "C:\bld\pdgraster_1719259170079\_test_env\lib\site-packages\pdgraster\__init__.py", line 3, in <module>
    from .WebImage import WebImage
  File "C:\bld\pdgraster_1719259170079\_test_env\lib\site-packages\pdgraster\WebImage.py", line 11, in <module>
    from . import logging_config
  File "C:\bld\pdgraster_1719259170079\_test_env\lib\site-packages\pdgraster\logging_config.py", line 8, in <module>
    handler = logging.FileHandler("/tmp/log.log")
  File "C:\bld\pdgraster_1719259170079\_test_env\lib\logging\__init__.py", line 1169, in __init__
    StreamHandler.__init__(self, self._open())
  File "C:\bld\pdgraster_1719259170079\_test_env\lib\logging\__init__.py", line 1201, in _open
    return open_func(self.baseFilename, self.mode,
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\tmp\\log.log
```

The noarch package will be installable on Windows and broken there. Idealy, we should not make it installable by using os-noarch. See https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies. Do you want to implement that or try to mint a new release upstream that doesn't have this issue?

**Comment by mfisher87** [comment]
This is recorded as an [issue in the upstream package](https://github.com/PermafrostDiscoveryGateway/viz-raster/issues/27), but myself and Rushiraj are not members of that team, we're just consumers at the moment. We expect to become more integrated with that team in October. 

For now, it makes sense to make this package not installable under Windows as you suggested, but I'm not sure I understand how from the link you shared. Can you clarify? I imagined that we would leave the library as installable but broken in Windows, because that is also the state if you install it with pip.

**Comment by ocefpaf** [comment]
> I imagined that we would leave the library as installable but broken in Windows, because that is also the state if you install it with pip.

Correct, Python noarch packages are built on Linux but installable in all platforms and, in this case, broken on Windows.

You need to  add `- __unix  # [unix]` to your run dependencies, making is not installable on Windows, and then add a `conda-forge.yml` file to the recipe directory like:

```yaml
noarch_platforms:
  - linux_64
  - osx_64
```

Telling it not to build on Windows.

**Comment by mfisher87** [comment]
Thanks for clarifying. I'll take that path. I'll push a commit to update the `meta.yaml`, and then do I wait until the feedstock is created to update the `conda-forge.yml`?

**Comment by ocefpaf** [comment]
> I wait until the feedstock is created to update the `conda-forge.yml`?

You can add it to the recipe directory and it will be copied over to the feedstock.

**Comment by mfisher87** [comment]
It's expected that this didn't affect the staged-recipes build?

**Pull request merged**